hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/WritableAirdropStore.java

```diff
@@ -0,0 +1,135 @@
+    /**
+     * The underlying data storage class that holds the Pending Airdrops data.
+     */
+    private final WritableKVState<PendingAirdropId, AccountAirdrop> airdropState;
+
+    /**
+     * Create a new {@link WritableAirdropStore} instance.
+     *
+     * @param states The state to use.
+     */
+    public WritableAirdropStore(
+            @NonNull final WritableStates states,
+            @NonNull final Configuration configuration,
+            @NonNull final StoreMetricsService storeMetricsService) {
+        super(states);
+        airdropState = states.get(AIRDROPS_KEY);
+
+        final long maxCapacity = configuration.getConfigData(TokensConfig.class).maxAllowedAirdrops();
+        final var storeMetrics = storeMetricsService.get(StoreMetricsService.StoreType.AIRDROP, maxCapacity);
+        airdropState.setMetrics(storeMetrics);
+    }
+
+    /**
+     * Persists a new {@link PendingAirdropId} with given {@link AccountAirdrop} into the state,
+     * as well as exporting its ID to the transaction receipt. If there is existing
+     * airdrop with the same id we add the value to the existing drop.
+     *
+     * @param airdropId    - the airdropId to be persisted.
+     * @param accountAirdrop - the account airdrop mapping for the given airdropId to be persisted.
+     */
+    public void put(@NonNull final PendingAirdropId airdropId, @NonNull final AccountAirdrop accountAirdrop) {
+        requireNonNull(airdropId);
+        requireNonNull(accountAirdrop);
+
+        if (!airdropState.contains(airdropId)) {
+            airdropState.put(airdropId, accountAirdrop);
+            return;
+        }
+
+        if (airdropId.hasFungibleTokenType()) {
+            var existingAccountAirdrop = requireNonNull(airdropState.get(airdropId));
+            var existingAirdropValue =
+                    requireNonNull(airdropState.get(airdropId)).pendingAirdropValue();
+            requireNonNull(existingAirdropValue);
+            requireNonNull(accountAirdrop.pendingAirdropValue());
+            var newValue = accountAirdrop.pendingAirdropValue().amount() + existingAirdropValue.amount();
+            var newAccountAirdrop = existingAccountAirdrop
+                    .copyBuilder()
+                    .pendingAirdropValue(
+                            PendingAirdropValue.newBuilder().amount(newValue).build())
+                    .build();
+            airdropState.put(airdropId, newAccountAirdrop);
+        }
+    }
+
+    /**
+     * Updates given {@link PendingAirdropId} with new {@link AccountAirdrop} value the state,
+     * as well as exporting its ID to the transaction receipt. If there is no existing
+     * airdrop with the same id do nothing.
+     *
+     * @param airdropId    - the airdropId to be updated.
+     * @param accountAirdrop - the account airdrop mapping for the given airdropId to be updated.
+     */
+    public void patch(@NonNull final PendingAirdropId airdropId, @NonNull final AccountAirdrop accountAirdrop) {
```

Comment:
This seems like its not used and doesn't do the same thing as defined in javadoc

Reply:
We use it if we wan't to overwrite the AccountAirdrop for given PendingAirdropId, lets say update the nextPendingAirdrop and previousPendingAirdrop, when claiming or canceling airdrops. But we don't want to execute the login in the put method where for fungible tokens if we put to a existing value += the pending airdrop value.